wire-compiler/src/main/java/com/squareup/wire/model/WireMessage.java

```diff
@@ -115,4 +94,28 @@ void link(Linker linker) {
       option.link(ProtoTypeName.MESSAGE_OPTIONS, linker);
     }
   }
+
+  @Override WireType retainAll(Set<String> identifiers) {
```

Comment:
I'm pretty sure there's a tiny simplification to be had by switching these methods to return collections. You never have to worry about null checking and have logic for one vs. many. It's always just either an `isEmpty()` check or `addAll` when composing.